lib/elixir/lib/gen_server.ex

```diff
@@ -565,15 +565,13 @@ defmodule GenServer do
   that are two-element tuples with a reference as the first element.
   """
   @spec call(server, term, timeout) :: term
-  def call(server, request, timeout \\ 5000)
-
-  def call(server, request, timeout) when server == self() do
-    exit({:calling_self, {__MODULE__, :call, [server, request, timeout]}})
-  end
-
-  def call(server, request, timeout) do
+  def call(server, request, timeout \\ 5000) do
+    pid = whereis(server)
+    if pid == self() do
+      exit({:calling_self, {__MODULE__, :call, [server, request, timeout]}})
+    end
```

Comment:
Like an extra if statement after this one or replacing it by a case statement or something?